lib/ansible/module_utils/ipa.py

```diff
@@ -68,23 +71,38 @@ def get_json_url(self):
         return '%s/session/json' % self.get_base_url()
 
     def login(self, username, password):
-        url = '%s/session/login_password' % self.get_base_url()
-        data = 'user=%s&password=%s' % (quote(username, safe=''), quote(password, safe=''))
-        headers = {'referer': self.get_base_url(),
-                   'Content-Type': 'application/x-www-form-urlencoded',
-                   'Accept': 'text/plain'}
-        try:
-            resp, info = fetch_url(module=self.module, url=url, data=to_bytes(data), headers=headers, timeout=self.timeout)
-            status_code = info['status']
-            if status_code not in [200, 201, 204]:
-                self._fail('login', info['msg'])
-
-            self.headers = {'referer': self.get_base_url(),
-                            'Content-Type': 'application/json',
-                            'Accept': 'application/json',
-                            'Cookie': resp.info().get('Set-Cookie')}
-        except Exception as e:
-            self._fail('login', to_native(e))
+        if 'KRB5_CLIENT_KTNAME' in os.environ and HAS_GSSAPI:
+            ccache = "MEMORY:" + str(uuid.uuid4())
+            os.environ['KRB5CCNAME'] = ccache
```

Comment:
Please do not override existing KRB5CCNAME. You should only use MEMORY: ccaches if no KRB5CCNAME is defined in the environment already.

Reply:
Good call! Let me update this.

Reply:
Updated it. Does this make sense?